<launch>
	<param name = "logging" value = "false" type = "bool" />
	<param name = "PCL2TopicName" value = "/duo3d/point_cloud/image_raw" type = "str" />

	<arg name="cameraConfig" default="$(find occupancy_grid)/cfg/occupancy_grid_config.yaml"/>
	<rosparam command="load" file="$(arg cameraConfig)"/>

	<node pkg="occupancy_grid" type="occupancy_grid" name="occupancy_grid">
	</node>

	<node name="duo3d" pkg="duo3d_driver" type="duo3d_driver" output="screen">
		<param name="frame_rate" value="20.0"/>
		<rosparam param="image_size">[320, 240]</rosparam>
		<param name="dense3d_license" value="OVMZU-ZHFE2-9K41K-NQL44-WX3DV"/>
		
		<param name="gain" value="0"/>
		<param name="exposure" value="0"/>
		<param name="auto_exposure" value="false"/>
		<param name="vertical_flip" value="false"/>
		<param name="horizontal_flip" value="false"/>

		<param name="processing_mode" value="1"/>
		<param name="image_scale" value="3"/>
		<param name="pre_filter_cap" value="8"/>
		<param name="num_disparities" value="4"/>
		<param name="sad_window_size" value="7"/>
		<param name="uniqueness_ratio" value="11"/>
		<param name="speckle_window_size" value="179"/>
		<param name="speckle_range" value="14"/>
    	</node>
</launch>
